ext/bcmath/libbcmath/src/recmul.c

```diff
@@ -38,23 +38,26 @@
 
 
 #if SIZEOF_SIZE_T >= 8
-#  define BC_MUL_UINT_DIGITS 8
-#  define BC_MUL_UINT_OVERFLOW (BC_UINT_T) 100000000
+#  define BC_VECTORIZED_MULTIPLICATION_VECTOR_SIZE 8
+/* The boundary number is computed from BC_VECTORIZED_MULTIPLICATION_VECTOR_SIZE times BASE */
+#  define BC_VECTORIZED_MULTIPLICATION_BOUNDARY_NUMBER (BC_INT_T) 100000000
+#  define BC_MUL_MAX_ADD_COUNT (((BC_INT_T) 0x0fffffffffffffff) / \
+		(BC_VECTORIZED_MULTIPLICATION_BOUNDARY_NUMBER * BC_VECTORIZED_MULTIPLICATION_BOUNDARY_NUMBER))
 #else
-#  define BC_MUL_UINT_DIGITS 4
-#  define BC_MUL_UINT_OVERFLOW (BC_UINT_T) 10000
+#  define BC_VECTORIZED_MULTIPLICATION_VECTOR_SIZE 4
+#  define BC_VECTORIZED_MULTIPLICATION_BOUNDARY_NUMBER (BC_INT_T) 10000
+#  define BC_MUL_MAX_ADD_COUNT (((BC_INT_T) 0x0fffffff) / \
+		(BC_VECTORIZED_MULTIPLICATION_BOUNDARY_NUMBER * BC_VECTORIZED_MULTIPLICATION_BOUNDARY_NUMBER))
```

Comment:
Shouldn't this also have a new name? I'm not sure what it should be called, but is `BC_VECTORIZED_MULTIPLICATION_MAXIMAL_ADDITION` correct?

Reply:
This is the threshold at which "adding more than this many times may cause int32_t/int64_t to overflow."

It might be better to clarify that this refers to the number of times, not the maximum value that can be added.

If I directly translate the expression I came up with in Japanese into English, it would be "The maximum number of times that can be added without overflowing" but this is too long and I'm not sure what the right wording is.

Reply:
`BC_VECTORIZED_MULTIPLICATION_NO_OVERFLOW_COUNT` ?

Reply:
Also, why is this `0x0fffffff`, can't this be `0x7fffffff` ? (and similar reasoning for the 64-bit one)

Reply:
> BC_VECTORIZED_MULTIPLICATION_NO_OVERFLOW_COUNT ?

If shorten BC_VECTORIZED_MULTIPLICATION_VECTOR_SIZE, you might want to shorten this a bit too.

> Also, why is this 0x0fffffff, can't this be 0x7fffffff ? (and similar reasoning for the 64-bit one)

Ah, you are right. I was trying to set the first bit to 0, but accidentally set all 4 bits to 0. I'll fix it